### DISCUSSION TITLE
offtopic: need help for mcu-uuid-console with telnet logging

### DISCUSSION THREAD

**Discussion by SpaceTeddy** [emsesp/EMS-ESP32#2289] [Q&A] [answered]
Hi everyone,
ems-esp32 uses the nice mcu-uuid-console, telnet, log and syslog library, which is really cool. Therefore I would like to use these libs also for my project, but I'm in trouble to activate the logging feature in the telnet console output. Simon's example's working fine and the console is running via telnet perfectly and commands are executed as expected. The example for Serial logging is also working, but I would like to log via telnet console. My workaround is currently the syslog lib, but this is a bit oversized for my project :)
Therefore I would like to ask for help if somebody may teach me to activate the logging in the telnet console.

thanks for help und understanding ;)

attached the basic sketch
```cpp
#include <Arduino.h>

#include <memory>
#include <string>
#include <vector>

#include <uuid/common.h>
#include <uuid/console.h>
#include <uuid/telnet.h>
#include <uuid/syslog.h>
#include "commands.h"

using uuid::read_flash_string;
using uuid::flash_string_vector;
using uuid::console::Commands;
using uuid::console::Shell;

// uuid-console + telnet
static std::shared_ptr<uuid::console::Commands> commands = std::make_shared<uuid::console::Commands>();
static uuid::telnet::TelnetService telnet{commands};

//uuid-syslog
static uuid::syslog::SyslogService syslog;

setup(){
    //Setup Console
    registerCommands(commands);
    telnet.start();

    //Setup syslog
    //Setup Logger
    static uuid::log::Logger logger{F("setup")};

    syslog.start();
    String hostname_syslog = hostname_base + get_flashmemory_id();
    syslog.hostname(hostname_syslog.c_str());
    syslog.log_level(uuid::log::ALL);
    syslog.mark_interval(3600);
    syslog.destination(IPAddress(192, 168, 0, 207), 514);

    // Log testmessage
    logger.info(F("LLU Client Application started"));
}

loop(){

    uuid::loop();
    syslog.loop();
    telnet.loop();
    Shell::loop_all();yield();

}
```

**Comment by proddy** [comment]
Take a look at `start_serial_console()` in the file `emsesp.cpp`. Once you create a log instance, which you did, it'll be exposed to all `Shell`s

**Comment by SpaceTeddy** [comment]
hey Proddy,
thank for answering.
o be honest, my c++ skills are not really good and I have no idea how to continue with all the namespaces and pointers and references ,)
May I ask you to give me some hints for my added POC example?
This would be really nice.

chris

**Reply by proddy** [reply]
create a public Github repo with the code, I'll fork it and fix it. When I have time!

**Reply by SpaceTeddy** [reply]
Ok,
Thanks a lot.
Attached the test report:
https://github.com/SpaceTeddy/uuid-console-log-telnet-test.git

I did not test this on a target, but it compiles well.

thanks a lot for your time :)

**Comment by SpaceTeddy** [comment, marked as answer]
dear Proddy,
no need for tsting it anymore. 
Simon gave me the hint, that the standard Loglevel in the Console is NOTICE and I tried with INFO, which is lower.
After changing the level to ALL with:
```cpp
void setLoglevelCommand(uuid::console::Shell &shell, const std::vector<std::string> &) {
    shell.printfln(F("Set Loglevel to ALL:"));
    shell.log_level(uuid::log::Level::ALL);
}
```
I can see the logging in the console as well.

sry for disturbing 

sers,
Chris
